packages/cadl-rlc-emitter/src/index.ts

```diff
@@ -1,55 +1,34 @@
-import { CompilerHost, Program } from '@cadl-lang/compiler';
-import { buildClientDefinitions, Paths } from '@azure-tools/rlc-codegen';
-import { getAllRoutes } from '@cadl-lang/rest/http';
-import { dirname, join } from 'path';
+import { CompilerHost, Program } from "@cadl-lang/compiler";
+import { buildClientDefinitions, Paths } from "@azure-tools/rlc-codegen";
+import { dirname, join } from "path";
+import { transformPaths } from "./transformPaths";
 
 export async function $onEmit(program: Program) {
-    const [routes, _diagnostics] = getAllRoutes(program);
-    const paths: Paths = {};
-    for (const route of routes) {
-        paths[route.path] = {
-            name: route.operation.name,
-            pathParameters: route.parameters.parameters
-                .filter((p) => p.type === 'path')
-                .map((p) => {
-                    return {
-                        name: p.name,
-                        type: 'string',
-                        description: 'param'
-                    };
-                }),
-            methods: {
-                [route.verb]: [
-                    {
-                        description: 'Foo',
-                        hasOptionalOptions: route.parameters.parameters.some(
-                            (p) => p.param.optional
-                        ),
-                        optionsName: 'options',
-                        responseTypes: { success: ['string'], error: [] },
-                        returnType: '',
-                        successStatus: ['200']
-                    }
-                ]
-            }
-        };
+  const paths: Paths = transformPaths(program);
+  const clientDefinitionsFile = buildClientDefinitions(
+    {
+      paths,
+      libraryName: "Foo",
+      srcPath: "src",
+      options: {
+        // TODO: Read configuration from CADL
+        includeShortcuts: true
+      }
+    },
+    {
+      clientImports: new Set(),
+      importedParameters: new Set(),
+      importedResponses: new Set()
     }
```

Comment:
the same here, I wonder if we should have an independent function emitClientDefinition and then call emitClientDefinition inside onEmit ?